baize/datastructures.py

```diff
@@ -445,12 +443,14 @@ class Headers(MultiMapping[str, str]):
     def __init__(
         self,
         headers: typing.Mapping[str, str] = None,
-        raw: typing.List[typing.Tuple[str, str]] = None,
+        raw: typing.Iterable[typing.Tuple[str, str]] = None,
         scope: Scope = None,
         environ: Environ = None,
     ) -> None:
         if headers is not None:
-            _items = ((key.lower(), value) for key, value in headers.items())
+            _items: typing.Optional[typing.Iterable[typing.Tuple[str, str]]] = (
```

Comment:
excess type annotations. mypy is not so stupid.

Reply:
If not annotated _items will be inferred as Generator type and later assignment to List will fail.

Reply:
Yes, it is like this in your code. This is why I clearly need a Sequence instead of an Iterable object.